apps/web/app/_utils.tsx

```diff
@@ -1,72 +1,103 @@
+  pl: () => import("@calcom/web/public/static/locales/pl/common.json").then((module) => module.default),
+  pt: () => import("@calcom/web/public/static/locales/pt/common.json").then((module) => module.default),
+  "pt-BR": () =>
+    import("@calcom/web/public/static/locales/pt-BR/common.json").then((module) => module.default),
+  ro: () => import("@calcom/web/public/static/locales/ro/common.json").then((module) => module.default),
+  ru: () => import("@calcom/web/public/static/locales/ru/common.json").then((module) => module.default),
+  sk: () => import("@calcom/web/public/static/locales/sk/common.json").then((module) => module.default),
+  "sk-SK": () =>
+    import("@calcom/web/public/static/locales/sk-SK/common.json").then((module) => module.default),
+  sr: () => import("@calcom/web/public/static/locales/sr/common.json").then((module) => module.default),
+  sv: () => import("@calcom/web/public/static/locales/sv/common.json").then((module) => module.default),
+  ta: () => import("@calcom/web/public/static/locales/ta/common.json").then((module) => module.default),
+  th: () => import("@calcom/web/public/static/locales/th/common.json").then((module) => module.default),
+  tr: () => import("@calcom/web/public/static/locales/tr/common.json").then((module) => module.default),
+  uk: () => import("@calcom/web/public/static/locales/uk/common.json").then((module) => module.default),
+  vi: () => import("@calcom/web/public/static/locales/vi/common.json").then((module) => module.default),
+  "zh-CN": () =>
+    import("@calcom/web/public/static/locales/zh-CN/common.json").then((module) => module.default),
+  "zh-TW": () =>
+    import("@calcom/web/public/static/locales/zh-TW/common.json").then((module) => module.default),
+};
 
-  const _i18n = i18next.createInstance();
-  await _i18n.init({
-    lng: locale,
-    resources: _nextI18Next?.initialI18nStore,
-    fallbackLng: _nextI18Next?.userConfig?.i18n.defaultLocale,
-  });
+type LocaleType = keyof typeof dictionaries;
+const translationCache = new Map<string, TFunction>();
+async function _getServerTranslation(locale: string) {
+  const dict = await dictionaries[locale as LocaleType]();
 
-  // Cache the instance
-  i18nInstanceCache[cacheKey] = _i18n;
-  return _i18n;
-};
+  return {
+    t: (key: string, interpolation?: Record<string, string | number>) => {
+      let value = dict[key as keyof typeof dict] as string;
+      if (interpolation) {
+        Object.entries(interpolation).forEach(([k, v]) => {
+          value = value.replace(`{{${k}}}`, String(v));
+        });
+      }
+
+      return value;
+    },
+  };
+}
 
-const getTranslationWithCache = async (locale: string, ns = "common") => {
-  const localeWithFallback = locale ?? "en";
-  const i18n = await createI18nInstance(localeWithFallback, ns);
-  return i18n.getFixedT(localeWithFallback, ns);
-};
+export const getTranslate = async (lang: string): Promise<TFunction> => {
```

Comment:
Considering the way we use this function `getTranslate(params.lang as string)`, is there a possibility where this `lang` is falsey? Do we need to fall back to en? or does the fallback happen at the next.js level and we always have lang here?

Reply:
Good point! addressed